ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/NotaryLookup.java

```diff
@@ -0,0 +1,47 @@
+package net.corda.v5.ledger.common;
+
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import net.corda.v5.base.types.MemberX500Name;
+import net.corda.v5.membership.NotaryInfo;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+/**
+ * Defines a mechanism to allow flows to retrieve the NotaryInfo in the network.
+ * The platform will provide an instance of NotaryLookup to flows via property injection.
+ */
+@DoNotImplement
+public interface NotaryLookup {
+
+    /**
+     * Gets a {@link Collection} of {@link NotaryInfo} services available on the network.
+     *
+     * @return Returns a {@link Collection} of {@link NotaryInfo} services available on the network.
+     */
+    @NotNull
+    @Suspendable
+    // TODO : Consider List<NotaryInfo>
+    Collection<NotaryInfo> getNotaryServices();
```

Comment:
If its suspendable, doesn't that imply it needs to be serializable, and IIRC aren't there issues with sets, maps etc. with serializing? Or am I getting confused with something else? 

Is there a reason not to leave it as a collection, given it's an interface? In practice I suspect it doesn't matter too much, given there's not likely to be too many notary services on a network anyway.

Reply:
@lankydan @relyafi I'll leave the TODO there for the time being. There are a few in the API for consideration, though we probably want to address them sooner rather than later if they're API concerns.

Reply:
If we're going to leave the TODO, we should raise a ticket for it and put it highest so it can get scoped. We either do this now or we don't. If we put it as a "should" it won't get done regardless.

Reply:
Fair point.

So going back to my original comment:

_Is there a reason not to leave it as a collection, given it's an interface?_

Can we start with answering that?